pagopa/io-app
# fix(Cross): [IOAPPX-391] Move `watchZendeskSupportSaga` to `root` saga

## Short description
This PR refactors the handling of the `watchZendeskSupportSaga`, moving it from the `startup` saga to the `root` saga. This ensures that Zendesk support is handled more consistently throughout the application lifecycle, especially during the FL refresh.

## List of changes proposed in this pull request
- Moved `watchZendeskSupportSaga` from `startup` saga to `root` saga for better navigation management and modal handling.

## How to test
1. Background test:
    - Start a session as an authenticated user.
    - Open the Zendesk modal.
    - Leave the application in the background for two minutes.
    - Bring the application back to the foreground and identify yourself.
    - Verify the automatic session refresh, then attempt to close the Zendesk modal.
    - Ensure the modal closes without issues.
> [!caution]
> **Note**: Remember to use "the-safe-word" while opening a ticket with Zendesk in production. (Reach out if you are unsure what this is 😉).
 2. Not authenticated session: 
     - Try to open a Zendesk ticket in a non-authenticated part of the application.
 3. Onboarding/First Onboarding:
    - Attempt to open a Zendesk ticket during an onboarding process.
 4. Authenticated session:
    - Open a ticket from an authenticated section of the application and verify behavior.

<details><summary>🎦 Demo</summary>
<p>

| ❌| ✅ | 
| - | - | 
| <video src="https://github.com/user-attachments/assets/923d395f-115e-4b04-8b67-d5d2fe6a56ee" /> | <video src="https://github.com/user-attachments/assets/02d4e9ed-faf1-4718-847e-ee8ffb0abeb4" /> |

</p>
</details>
